cockroach-metrics/src/lib.rs

```diff
@@ -23,14 +24,30 @@ use tokio::sync::RwLock;
 ///
 /// Only accesses a single client at a time. To query from multiple nodes
 /// in a cluster concurrently, use [CockroachClusterAdminClient].
+#[derive(Clone)]
 struct CockroachAdminClient {
     client: Client,
 }
 
 impl CockroachAdminClient {
     /// Create a new CockroachDB HTTP client
     fn new(log: Logger, address: SocketAddr) -> Self {
-        let client = Client::new(&format!("http://{address}"), log);
+        // It's important that we have *some* timeout here - currently,
+        // inventory collection will query all nodes to confirm they're
+        // responding. However, it's very possible that one node is down,
+        // and that should not block collection indefinitely.
+        let timeout_duration = std::time::Duration::from_secs(15);
```

Comment:
This is maybe an unreasonable fear, but - if any non-inventory bits want to start talking to the cockroach admin server and reach for this, will they get a potentially-surprising timeout set? Would it be reasonable to take `timeout_duration` as a parameter to new and force the user of the client to pick?

Reply:
Done in eef1ee1c071e765ea101102307d55c6e3e36811f